src/ws.cpp

```diff
@@ -348,27 +348,30 @@ bool Instance::initialize(EGLDisplay eglDisplay)
         return false;
     }
 
+    if (wl_display_init_shm(m_display))
```

Comment:
I find this a bit confusing, it seems like we return false when init shm succeeds, it's not obvious that it returns 0 for success and -1 for failure. So, I would explicitly check == -1 or != 0